### PR TITLE
Update oidc/jose library #17 #8

### PR DESCRIPTION
**Pull request by p53**
Update of oidc/jose libraries, turn off client id check for verifying access token, related to https://github.com/gogatekeeper/gatekeeper/issues/8, https://github.com/gogatekeeper/gatekeeper/issues/17

### CODE REVIEW

**Comment by p53** [comment]
- logout Handler - revocation endpoint needs to be fixed, remove old oidc.ProviderConfig from oauthProxy struct

**Comment by p53** [comment]
iconfinder_christmas_tree_62123

**Comment by p53** [comment]
@fredbi @ackerleytng @statik check it out, tests pass, i tried it on simple use cases in my project, needs to be tested - best would be integration tests

File: forwarding.go [hunk 1]
```diff
@@ -185,13 +195,12 @@ func (r *oauthProxy) forwardProxyHandler() func(*http.Request, *http.Response) {
...
-						zap.String("subject", state.identity.ID),
-						zap.String("email", state.identity.Email),
+						zap.String("subject", state.subject),
 						zap.String("expires", state.expiration.Format(time.RFC3339)))
 				} else {
 					r.log.Info("session does not support refresh token, acquiring new token",
-						zap.String("subject", state.identity.ID),
-						zap.String("email", state.identity.Email))
+						zap.String("subject", state.subject),
+						zap.String("email", state.subject))
```
Comment (ackerleytng) [inline forwarding.go:203]: Did you mean to remove this part about the email, like for the other cases?

File: doc.go [hunk 2]
```diff
@@ -433,7 +431,11 @@ type userContext struct {
 	// roles is a collection of roles the users holds
 	roles []string
 	// the access token itself
-	token jose.JWT
+	token *jwt.JSONWebToken
```
Comment (ackerleytng) [inline doc.go:425]: If this `token` field isn't used anymore (with the introduction of `rawToken`), perhaps we can just remove the token field (or replace the `token` field with the `rawToken` string.

File: session.go [hunk 3]
```diff
@@ -21,32 +21,37 @@ import (
...
 	"strings"
 
-	"github.com/coreos/go-oidc/jose"
 	"go.uber.org/zap"
+	"gopkg.in/square/go-jose.v2/jwt"
 )
 
 // getIdentity retrieves the user identity from a request, either from a session cookie or a bearer token
-func (r *oauthProxy) getIdentity(req *http.Request) (*userContext, error) {
+func (r *oauthProxy) getIdentity(req *http.Request) (*userContext, string, error) {
```
Comment (ackerleytng) [inline session.go:29]: Seems like `rawToken` is already sort of returned through `userContext`. I feel that it is neater not to have this function do more than promised by also returning the `rawToken` as one of the return values.

File: middleware.go [hunk 4]
```diff
@@ -197,6 +197,7 @@ func (r *oauthProxy) authenticationMiddleware() func(http.Handler) http.Handler
 					&oidc3.Config{
 						ClientID:          r.config.ClientID,
 						SkipClientIDCheck: true,
+						SkipIssuerCheck:   true,
```
Comment (ackerleytng) [inline middleware.go:200]: I'm concerned about existing users that rely on this issuer check for security. Perhaps we can add a config flag that allows `SkipIssuerCheck`, but default it to `false`? Or maybe we need to discuss how we want to approach backward compatibility

Comment (p53) [reply middleware.go:200]: yup, we can do that altough this checking is in fact bug, we can provide flags to have users more fluent migration

Comment (p53) [reply middleware.go]: i added flags

File: session.go (hunk 3 above)
Comment (p53) [reply session.go]: fixed

File: forwarding.go (hunk 1 above)
Comment (p53) [reply forwarding.go:203]: state doesn't have identity after refactor and even isn't in standard claims thus and this info is just for logging, doesn't seem to have big importance, i will remove it from logging

**Comment by p53** [comment]
@fredbi @ackerleytng @statik i fixed issues in review, if there are no other comments i would like to make 1.2.0-rc-1

**Pull request merged**